cluster-api-gcp/deploy/deploy_helper.go

```diff
@@ -319,19 +326,24 @@ func mergeConfigIfExists(newConfig, existingFilePath string) (*clientcmdapi.Conf
 }
 
 // Make sure the default service account in kube-system namespace exists.
-func (d *deployer) waitForServiceAccount(timeout time.Duration) error {
+func (d *deployer) waitForServiceAccount() error {
 	client, err := apiutil.NewKubernetesClient(d.configPath)
 	if err != nil {
 		return err
 	}
-	startTime := time.Now()
 
-	for time.Now().Sub(startTime) < timeout {
-		_, err := client.CoreV1().ServiceAccounts(ServiceAccountNs).Get(ServiceAccountName, metav1.GetOptions{})
+	waitErr := util.Retry(func() (bool, error) {
+		glog.Info("Waiting for the service account to exist...")
+		_, err = client.CoreV1().ServiceAccounts(ServiceAccountNs).Get(ServiceAccountName, metav1.GetOptions{})
 		if err == nil {
```

Comment:
Perhaps cleaner to next the bool evaluation since you do not actually do anything special with the err.

return (err == nil), nil

Same in many other functions.

Reply:
Done.